core/shared/src/main/scala/zio/FiberId.scala

```diff
@@ -31,6 +29,7 @@ sealed trait FiberId extends Serializable { self =>
 
   final def combine(that: FiberId): FiberId =
     (self, that) match {
+      case (None, None) => None
```

Comment:
By adding this condition we know we can't end-up with a `Composite(None, None)` which means that `isNone` for a `Composite` is always `false`

Reply:
`Composite` is not private though, so maybe it could be constructed by something else?

It's still good to add `case (None, None) => None`

Reply:
Good point. I'm rusted ahah.

Should we make these case classes `private[zio]`? Any reason to let them public?

Reply:
I think we need to keep them for backward compatibility and potentially make them private if there's a ZIO 3.0 😭

Reply:
In this case we have two choices I think

1. We consider no one is building (should be building) a `Composite` manually in its own code and so we can keep the optimised version of `isNone` as proposed. 
2. We go for a slightly less optimized but safe version:
```scala
  final def isNone: Boolean =
    self match {
      case None                     => true
      case FiberId.Runtime(_, _, _) => false
      case Composite(_, _)          => toSet.forall(_.isNone)
    }
```

I vote for solution 1. 🙂

Reply:
I don't see any usage of `getOrElse` or `isNone` outside this file, so the optimization is not really beneficial. In that case I think it's better to keep the correctness.